virtcontainers/kata_agent.go

```diff
@@ -843,6 +848,43 @@ func (k *kataAgent) handleBlockVolumes(c *Container) []*grpc.Storage {
 	return volumeStorages
 }
 
+func (k *kataAgent) handlePidNamespace(grpcSpec *grpc.Spec, sandbox *Sandbox) (bool, error) {
+	sharedPidNs := false
+
+	for i, ns := range grpcSpec.Linux.Namespaces {
+		if ns.Type == string(specs.PIDNamespace) {
+			if ns.Path != "" {
+				if sandbox.state.Pid > 0 {
+					pidNsPath := fmt.Sprintf("/proc/%d/ns/pid", sandbox.state.Pid)
+
+					//  Check if pid namespace path is the same as the sandbox
+					if ns.Path == pidNsPath {
+						sharedPidNs = true
+					} else {
+						// Check for symlinks
+						ln, err := filepath.EvalSymlinks(ns.Path)
+						if err != nil {
+							return sharedPidNs, err
+						}
+
+						// We have arbitrary pid namespace path here
+						if ln != pidNsPath {
+							return sharedPidNs, fmt.Errorf("Pid namespace path other than sandbox")
+						}
+						sharedPidNs = true
+					}
+				}
+			}
+
+			// Remove pid namespace.
+			grpcSpec.Linux.Namespaces = append(grpcSpec.Linux.Namespaces[:i], grpcSpec.Linux.Namespaces[i+1:]...)
```

Comment:
Whenever I see this standard idiom, I always have to do a double-take as it just looks wrong to me :)